airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/backoff_strategy.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import requests
+
+
+class BackoffStrategy(ABC):
+    @property
+    @abstractmethod
+    def max_retries(self) -> int:
+        """
+        Override if needed. Specifies maximum amount of retries for backoff policy. Return None for no limit.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def max_time(self) -> int:
+        """
+        Override if needed. Specifies maximum total waiting time (in seconds) for backoff policy. Return None for no limit.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def retry_factor(self) -> float:
```

Comment:
This method shouldn't be part of the public interface because it only makes sense for exponential backoffs. A strategy that backoffs with a constant time or extracts the time from a response header wouldn't have a retry factor

Reply:
Updated